pkg/binding/controller.go

```diff
@@ -267,6 +280,23 @@ func (c *Controller) run(ctx context.Context, workers int) error {
 	return nil
 }
 
+func (c *Controller) createManagedClustersInformer(ctx context.Context) {
+	informerFactory := ocm.GetOCMInformerFactory(c.ocmClientset)
+	informer := informerFactory.Cluster().V1().ManagedClusters().Informer()
+	_, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		// TODO: we can be smarter on how we handle at least the delete rather than
+		// recreate all the bindings.
+		AddFunc:    func(obj interface{}) { c.evaluateBindingPolicies(ctx) },
```

Comment:
same comment as delete.. you can check labels of the MC and re-evaluate only relevant placements (ones that selects labels that exist in this MC).

Reply:
Done